modules/scratchFetch.ts

```diff
@@ -0,0 +1,218 @@
+import { defineEvent } from "strife.js";
+import fetch from "node-fetch";
+import { getSettings } from "./settings.js";
+
+defineEvent("messageCreate", async (message) => {
+	const notSet = getSettings(message.author, false)?.scratchEmbeds === undefined;
+	if (!getSettings(message.author, false).scratchEmbeds && !notSet) {
+		return;
+	}
+	const scratchUrlRegex =
+		/(?<!<)https?:\/\/scratch\.mit\.edu\/(projects|users|studios)\/\w+\/?(?!>)/; //gpt wrote the regex and like half of this code
```

Comment:
Bugs with the regex
- Usernames containing dashes are not matched
- If there's a `<` before the link **and/or** after it will fail
- The embed should still be created if there's more stuff after the `/`, for instance `#comments-12345` or `/remixes`